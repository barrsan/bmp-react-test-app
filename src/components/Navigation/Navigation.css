.navigation {
  padding: 10px;
}

.nav-link.active {
  background-color: gainsboro;
  border-radius: 10px;
}