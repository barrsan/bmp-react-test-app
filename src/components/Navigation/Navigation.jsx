import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navigation.css';


const Navigation = () => (
  <div className="navigation">
    <ul className="nav">
      <li className="nav-item">
        <NavLink
          className="nav-link"
          exact
          to="/"
        >
          Экспонаты
        </NavLink>
      </li>
      <li className="nav-item">
        <NavLink
          className="nav-link"
          to="/new"
        >
          Добавить экспонат
        </NavLink>
      </li>
    </ul>
  </div>
);


export default Navigation;
