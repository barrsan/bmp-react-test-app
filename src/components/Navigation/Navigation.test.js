import React from 'react';
import { shallow } from 'enzyme';
import Navigation from './Navigation';


describe('Navigation component', () => {
  const wrapper = shallow(<Navigation />);

  test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  test('renders without crashing', () => {
    expect(wrapper).toHaveLength(1);
  });
});
