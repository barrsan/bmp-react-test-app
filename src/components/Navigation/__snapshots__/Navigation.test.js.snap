// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="nav"
>
        <li
                className="nav-item"
        >
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/"
                >
                        Экспонаты
                </NavLink>
        </li>
        <li
                className="nav-item"
        >
                <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        to="/new"
                >
                        Добавить экспонат
                </NavLink>
        </li>
</ul>,
      "className": "navigation",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="nav-item"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        exact={true}
                        to="/"
            >
                        Экспонаты
            </NavLink>
</li>,
          <li
            className="nav-item"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="nav-link"
                        to="/new"
            >
                        Добавить экспонат
            </NavLink>
</li>,
        ],
        "className": "nav",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              exact={true}
              to="/"
>
              Экспонаты
</NavLink>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Экспонаты",
              "className": "nav-link",
              "exact": true,
              "to": "/",
            },
            "ref": null,
            "rendered": "Экспонаты",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-link"
              to="/new"
>
              Добавить экспонат
</NavLink>,
            "className": "nav-item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Добавить экспонат",
              "className": "nav-link",
              "to": "/new",
            },
            "ref": null,
            "rendered": "Добавить экспонат",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="nav"
>
          <li
                    className="nav-item"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav-link"
                              exact={true}
                              to="/"
                    >
                              Экспонаты
                    </NavLink>
          </li>
          <li
                    className="nav-item"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="nav-link"
                              to="/new"
                    >
                              Добавить экспонат
                    </NavLink>
          </li>
</ul>,
        "className": "navigation",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="nav-item"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav-link"
                            exact={true}
                            to="/"
              >
                            Экспонаты
              </NavLink>
</li>,
            <li
              className="nav-item"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="nav-link"
                            to="/new"
              >
                            Добавить экспонат
              </NavLink>
</li>,
          ],
          "className": "nav",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                exact={true}
                to="/"
>
                Экспонаты
</NavLink>,
              "className": "nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Экспонаты",
                "className": "nav-link",
                "exact": true,
                "to": "/",
              },
              "ref": null,
              "rendered": "Экспонаты",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-link"
                to="/new"
>
                Добавить экспонат
</NavLink>,
              "className": "nav-item",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Добавить экспонат",
                "className": "nav-link",
                "to": "/new",
              },
              "ref": null,
              "rendered": "Добавить экспонат",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
